### PR TITLE
Fixed bug with isNotPresent in partial matcher.

### LINKED ISSUE #142: `isNotPresent` matcher not working correctly

### PR DESCRIPTION
**Pull request by anachlas**
Fixes #142

### CODE REVIEW

File: lib/src/utilities/test_matchers.dart [hunk 1]
```diff
@@ -379,18 +379,17 @@ class _PartialMapMatcher extends Matcher {
...
       var matchValue = map[matchKey];
       var value = item[matchKey];
-
-      if (value != null && matchValue is _NotPresentMatcher) {
-        var extra = matchState["extra"];
-        if (extra == null) {
-          extra = [];
-          matchState["extra"] = extra;
+      if (matchValue is _NotPresentMatcher) {
+        if (value != null) {
```
Comment (joeconwaystk) [inline lib/src/utilities/test_matchers.dart]: check that item.containsKey(matchKey) == false

File: test/utilities/test_client_test.dart [hunk 2]
```diff
@@ -492,6 +492,7 @@ void main() {
       var response = await defaultTestClient.request("/foo").get();
       expect(response, hasBody(partial({"foo": "bar"})));
       expect(response, hasBody(partial({"x": greaterThan(0)})));
+      expect(response, hasBody(partial({"baz": isNotPresent})));
```
Comment (joeconwaystk) [inline test/utilities/test_client_test.dart]: split into own test, add test for map that contains {"key" : null}

File: test/utilities/test_client_test.dart [hunk 3]
```diff
@@ -512,6 +512,33 @@ void main() {
         expect(e.toString(), contains('Body: {"foo":"bar","x":5}'));
       }
     });
+
+    test("Partial match, null and not present", () async {
```
Comment (joeconwaystk) [inline test/utilities/test_client_test.dart]: Is there a success case as well? expect(response, hasBody(partial({"abcd" : isNotPresent}))

Comment (anachlas) [inline test/utilities/test_client_test.dart]: I had it in the other test and I took it out. One sec.
